fhir-server/src/test/java/au/csiro/pathling/fhirpath/operator/CombineOperatorTest.java

```diff
@@ -0,0 +1,243 @@
+        .idAndEidAndValueColumns()
+        .expression("valueInteger")
+        .singular(false)
+        .build();
+    final IntegerLiteralPath right = IntegerLiteralPath
+        .fromString("99", parserContext.getInputContext());
+
+    final OperatorInput combineInput = new OperatorInput(parserContext, left, right);
+    final FhirPath result = Operator.getInstance("combine").invoke(combineInput);
+
+    final Dataset<Row> expectedDataset = new DatasetBuilder(spark)
+        .withIdColumn()
+        .withColumn(DataTypes.IntegerType)
+        .withRow("observation-1", 3)
+        .withRow("observation-1", 5)
+        .withRow("observation-1", 7)
+        .withRow("observation-1", 99)
+        .withRow("observation-2", null)
+        .withRow("observation-2", 99)
+        .withRow("observation-3", -1)
+        .withRow("observation-3", 99)
+        .build();
+    assertThat(result)
+        .hasExpression("valueInteger combine 99")
+        .isNotSingular()
+        .isElementPath(IntegerPath.class)
+        .selectResult()
+        .hasRowsUnordered(expectedDataset);
+  }
+
+  @Test
+  void worksWithLiteralAndNonLiteralCodingValues() {
+    final Dataset<Row> leftDataset = new DatasetBuilder(spark)
+        .withIdColumn()
+        .withEidColumn()
+        .withStructTypeColumns(codingStructType())
+        .withRow("observation-1", makeEid(0),
+            rowFromCoding(new Coding("http://snomed.info/sct", "18001011000036104", null)))
+        .buildWithStructValue();
+    final ElementPath left = new ElementPathBuilder(spark)
+        .fhirType(FHIRDefinedType.CODING)
+        .dataset(leftDataset)
+        .idAndEidAndValueColumns()
+        .expression("valueCoding")
+        .singular(false)
+        .build();
+    final CodingLiteralPath right = CodingLiteralPath
+        .fromString("http://snomed.info/sct|373882004", parserContext.getInputContext());
+
+    final OperatorInput combineInput = new OperatorInput(parserContext, left, right);
+    final FhirPath result = Operator.getInstance("combine").invoke(combineInput);
+
+    final Dataset<Row> expectedDataset = new DatasetBuilder(spark)
+        .withIdColumn()
+        .withStructTypeColumns(codingStructType())
+        .withRow("observation-1",
+            rowFromCoding(new Coding("http://snomed.info/sct", "18001011000036104", null)))
+        .withRow("observation-1",
+            rowFromCoding(new Coding("http://snomed.info/sct", "373882004", null)))
+        .withRow("observation-2",
+            rowFromCoding(new Coding("http://snomed.info/sct", "373882004", null)))
+        .withRow("observation-3",
+            rowFromCoding(new Coding("http://snomed.info/sct", "373882004", null)))
+        .buildWithStructValue();
+    assertThat(result)
+        .hasExpression("valueCoding combine http://snomed.info/sct|373882004")
+        .isNotSingular()
+        .isElementPath(CodingPath.class)
+        .selectResult()
+        .hasRowsUnordered(expectedDataset);
+  }
+
```

Comment:
Add test for combining resource paths.

Reply:
Add a ParserTest in 5610e894a23b60c0b2e3143109c7006d9a5022d8.